script.js

```diff
@@ -0,0 +1,209 @@
+    const result = await fetch(`http://localhost:8080/tasks/text/${element._id}`, {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+        'Access-Control-Allow-Origin': '*'
+      },
+      body: JSON.stringify({
+        text: input.value
+      })
+    });
+    const data = await result.json();
+
+    allTasks.forEach(item => {
+      if (item._id === element._id) {
+        console.log(data.text);
+        item.text = data.text;
+      }
+    });
+    render();
+  } catch (error) {
+    showError('Ошибка удаления задачи');
+  }
+};
+
+const editTask = (element) => {
+  indexEditableTask = element._id;
+  render(indexEditableTask);
+};
+
+const showError = (error) => {
+  const window = document.createElement('p');
+  const bodyContainer = document.getElementById('forError');
+  const body = document.getElementsByTagName('body')[0];
+
+  window.innerText = error;
+  bodyContainer.className = 'container_error container';
+  window.className = 'error';
+  body.append(window);
+  render();
+};
+
+const render = (indexEditableElement = null) => {
+  const allTasks = document.getElementById('list_with_Tasks');
+  while (allTasks.firstChild) {
+    allTasks.removeChild(allTasks.firstChild);
+  };
+
+  const copyAllTasks = [...allTasks];
+  copyAllTasks.sort((task, nextTask) => (task.isCheck < nextTask.isCheck) ? -1 : 1);
+  copyAllTasks.forEach(element => {
+    const task = document.createElement('div');
+    const elementsInTask = document.createElement('div');
+    const editingElements = document.createElement('div');
+    const checkBox = document.createElement('input');
+    const p = document.createElement('p');
+    const imageDeletingTask = document.createElement('img');
+    const imageEditingTask = document.createElement('img');
+
+    p.innerText = element.text;
+    checkBox.type = 'checkBox';
+    checkBox.checked = element.isCheck;
+    imageDeletingTask.src = './images/delete.svg';
+    imageEditingTask.src = './images/edit.svg';
+    task.className = 'lists_with_checkBox_container'
+
+    imageDeletingTask.onclick = () => deleteTask(element);
+    imageEditingTask.onclick = () => editTask(element);
+    checkBox.onclick = () => changeCheckBoxTask(element);
+
+    if (element.isCheck) {
+      editingElements.className = 'checkBox__active';
+    };
+
+    if (element._id === indexEditableElement) {
+      const actionCancelAndAdding = document.createElement('div');
+      const newText = document.createElement('input');
+      const imageCancel = document.createElement('img');
+      const imageAdding = document.createElement('img');
+
+      newText.className = 'lists_with_checkBox__input_text';
+      newText.value = element.text;
+      imageCancel.src = './images/cancel.png';
+      imageAdding.src = './images/apply.png';
+
+      imageAdding.onclick = () => changeTextTask(element);
+      imageCancel.onclick = () => render();

```

Comment:
imageCancel.onclick = () => render(); 
it is not correct to do so, there should be a separate function for canceling editing, and not calling render and updating the whole page. You only work with one element